pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -193,6 +193,11 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(proxy *model.Proxy,
 			// create default cluster
 			discoveryType := convertResolution(proxy, service.Resolution)
 			clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
+			if features.FilterGatewayClusterConfig && proxy.Type == model.Router && !proxy.GatewayClusterNames[clusterName] {
+				if service.Attributes.Namespace != "istio-system" {
```

Comment:
add a comment why we skip istio-system

Reply:
Also you can't hardcode istio-system, it can be installed in any namespace

Reply:
Yeah. And come to think of it, with mixerv2, we wont even need this special casing. Assuming that telemetry lite is the default.